bin/crew

```diff
@@ -486,20 +486,13 @@ def files (pkgName)
 end
 
 def whatprovides (regexPat)
-  fileArray = []
-  needle = regexPat.gsub(/-/,',').gsub(/,/,'\-')
-  Dir[CREW_META_PATH + '*.filelist'].each do |packageList|
-    packageName = File.basename packageList, '.filelist'
-    File.readlines(packageList).each do |line|
-      found = line[/#{needle}/] if line.ascii_only?
-      if found
-        fileLine = packageName + ': ' + line
-        unless fileArray.include? fileLine
-          fileArray.push(fileLine)
-        end
-      end
-    end
+  # Use grep version command to ascertain whether we have a working grep.
+  unless system('grep -V > /dev/null 2>&1')
+    abort 'Grep is not working. Please install it with \'crew install grep\''.lightred
   end
+  fileArray = []
+  @grepresults =  %x[grep "#{regexPat}" #{CREW_META_PATH}*.filelist].chomp.gsub('.filelist','').gsub(':',': ').gsub(CREW_META_PATH,'').split(/$/).map(&:strip)
```

Comment:
Does grep use regex patterns with out the `-e` option?

Reply:
`-e` is not necessary if the first argument given is the regex, and that argument is quoted.

Reply:
(As per the grep man page...)